lib/palindrome_check.rb

```diff
@@ -1,7 +1,38 @@
 # A method to check if the input string is a palindrome.
 # Return true if the string is a palindrome. Return false otherwise.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: It is O(n)/linear because the while loop is passing through each index position only once.
+# Space complexity:  O(1) because I’m not using any memory storage in this algorithm.
+
 def palindrome_check(my_phrase)
-  raise NotImplementedError
+  return false if my_phrase == nil
+  leftindex = 0
+  rightindex = my_phrase.length - 1
+
+  while leftindex < rightindex
+    l_char = my_phrase[leftindex]
+    r_char = my_phrase[rightindex]
+
+    if l_char == " " || l_char == nil
```

Comment:
The outer if check here is unnecessary. The character cannot be nil in a valid string, and the whitespace check is redundant with the until statement below.